nebula/ui/components/VPNComposerView.qml

```diff
@@ -0,0 +1,169 @@
+                        modelData instanceof VPNComposerBlockUnorderedList) {
+                    switch(composer.view) {
+                    case VPNComposerView.View.Guide:
+                        return VPNTheme.theme.listSpacing * 0.5
+                    case VPNComposerView.View.Message:
+                    default:
+                        return VPNTheme.theme.vSpacingSmall
+                    }
+                }
+
+                if (modelData instanceof VPNComposerBlockButton) {
+                    if(repeater.isFirstButton) {
+                        return VPNTheme.theme.vSpacingSmall * 2
+                    }
+                    return VPNTheme.theme.vSpacingSmall
+                }
+            }
+
+            Component {
+                id: titleBlock
+
+                VPNBoldInterLabel {
+                    text: loader.composerBlock.title
+                    font.pixelSize: VPNTheme.theme.fontSize
+                    lineHeight: VPNTheme.theme.labelLineHeight
+                    verticalAlignment: Text.AlignVCenter
+                    wrapMode: Text.Wrap
+                }
+            }
+
+            Component {
+                id: textBlock
+
+                VPNInterLabel {
+                    anchors.left: parent.left
+                    anchors.right: parent.right
+
+                    text: loader.composerBlock.text
+                    font.pixelSize: VPNTheme.theme.fontSizeSmall
+                    color: VPNTheme.theme.fontColor
+                    horizontalAlignment: Text.AlignLeft
+                }
+            }
+
+            Component {
+                id: listBlock
+
+                VPNInterLabel {
+                    property string listType: loader.composerBlock && composerBlock.type === "olist" ? "ol" : "ul"
+                    property var tagsList: loader.composerBlock.subBlocks.map(subBlock => `<li>${subBlock}</li>`)
+
+                    anchors.left: parent.left
+                    anchors.right: parent.right
+
+                    text: `<${listType} style='margin-left: -24px;-qt-list-indent:1;'>%1</${listType}>`.arg(tagsList.join(""))
+                    textFormat: Text.RichText
+                    font.pixelSize: VPNTheme.theme.fontSizeSmall
+                    color: VPNTheme.theme.fontColor
+                    horizontalAlignment: Text.AlignLeft
+                    Accessible.name: tagsList.join(".\n").replace(repeater.indentTagRegex, "")
+                    lineHeight: 20
+                }
+            }
+
+            Component {
+                id: buttonBlock
+
+                VPNButton {
+                    anchors.left: parent.left
+                    anchors.right: parent.right
+
+                    implicitHeight: VPNTheme.theme.rowHeight
+                    text: loader.composerBlock.text
+                    colorScheme: loader.composerBlock.style === VPNComposerBlockButton.Primary ? VPNTheme.theme.blueButton : VPNTheme.theme.redButton
```

Comment:
What about the other button types? Should we remove the unsupported types?

Reply:
What other button types? We only have primary, destructive, and link. Primary and destructive are handled here and link is a different component.